persistence/src/test/java/edu/unc/lib/dl/persist/services/edit/EditTitleServiceTest.java

```diff
@@ -0,0 +1,182 @@
+    private BinaryObject binaryObj;
+    @Mock
+    private OperationsMessageSender operationsMessageSender;
+    @Mock
+    private MODSValidator modsValidator;
+
+    @Captor
+    private ArgumentCaptor<Collection<PID>> pidsCaptor;
+
+    private EditTitleService service;
+    private PID pid;
+    private Document document;
+
+    @Before
+    public void init() throws Exception {
+        initMocks(this);
+
+        pid = PIDs.get(UUID.randomUUID().toString());
+        service = new EditTitleService();
+
+        service.setAclService(aclService);
+        service.setModsValidator(modsValidator);
+        service.setRepoObjLoader(repoObjLoader);
+        service.setOperationsMessageSender(operationsMessageSender);
+
+        when(repoObjLoader.getRepositoryObject(eq(pid))).thenReturn(contentObj);
+        when(contentObj.getDescription()).thenReturn(binaryObj);
+        when(aclService.hasAccess(any(PID.class), any(AccessGroupSet.class), eq(Permission.editDescription)))
+                .thenReturn(true);
+        when(repoObjLoader.getRepositoryObject(any(PID.class))).thenReturn(contentObj);
+        when(agent.getPrincipals()).thenReturn(groups);
+        when(agent.getUsername()).thenReturn("agentname");
+        when(operationsMessageSender.sendUpdateDescriptionOperation(anyString(), any(Collection.class))).thenReturn("message_id");
+
+        document = new Document();
+    }
+
+    @Test
+    public void editTitleTest() throws Exception {
+        String title = "new title";
+        document.addContent(new Element("mods", MODS_V3_NS)
+                .addContent(new Element("titleInfo", MODS_V3_NS)
+                        .addContent(new Element("title", MODS_V3_NS).setText("original title"))));
+        when(binaryObj.getBinaryStream()).thenReturn(convertDocumentToStream(document));
+
+        service.editTitle(agent, pid, title);
+
+        verify(operationsMessageSender).sendUpdateDescriptionOperation(anyString(), pidsCaptor.capture());
+        Collection<PID> pids = pidsCaptor.getValue();
+        assertEquals(pids.size(), 1);
+        assertTrue(pids.contains(pid));
+    }
+
+    @Test(expected = AccessRestrictionException.class)
+    public void insufficientAccessTest() throws Exception {
+        String title = "new title";
+        document.addContent(new Element("mods", MODS_V3_NS)
+                .addContent(new Element("titleInfo", MODS_V3_NS)
+                        .addContent(new Element("title", MODS_V3_NS).setText("original title"))));
+        when(binaryObj.getBinaryStream()).thenReturn(convertDocumentToStream(document));
+
+        doThrow(new AccessRestrictionException()).when(aclService)
+                .assertHasAccess(anyString(), eq(pid), any(AccessGroupSet.class), any(Permission.class));
+
+        service.editTitle(agent, pid, title);
+    }
+
+    @Test
+    public void noModsTest() throws Exception {
+        String title = "new title";
+        when(binaryObj.getBinaryStream()).thenReturn(null);
```

Comment:
i think you want to do `when(contentObj.getDescription()).thenReturn(null)` to be closer to the expected behavior. The last stubbing wins, so there's no problem overriding the default from the test init method.

Reply:
What would the expected behavior be in this case?  An empty document?

Reply:
Do you mean in terms of the return value from contentObj.getDescription, binaryObj.getBinaryStream, or what should the resulting that gets added to the record be? For the first, it should return null. For the second, it shouldn't reach that. And for the third it should create a new mods document with just the title in it.